dialogue-core/src/main/java/com/palantir/dialogue/core/ClientPool.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.palantir.dialogue.Channel;
+import java.io.Closeable;
+import java.io.IOException;
+
+/**
+ * Facilitates creating many clients which all share the same connection pool. Should only create one of these per
+ * server. Close it when your server shuts down to release resources.
+ */
+public interface ClientPool extends Closeable {
+
+    /** Returns an implementation of the given dialogueInterface, hooked up to a fresh smart channel underneath. */
+    <T> T get(Class<T> dialogueInterface, Listenable<DialogueConfig> config);
+
+    /**
+     * Returns a channel for interacting with the given abstract upstream service, which routes traffic
+     * appropriately to the various available nodes. Live-reloaded every time the urls change.
+     */
+    Channel smartChannel(Listenable<DialogueConfig> config);
```

Comment:
I think we should have some API separation between channel creation and using a channel to create a generated client (which would delegate to the first component)